File: examples/03-intro-java/Example7.java
// =================================================================
//
// File: Example7.java
// Author(s):
// Description: This file contains the code to brute-force all
//				prime numbers less than MAXIMUM. The time this
//				implementation takes will be used as the basis to
//				calculate the improvement obtained with parallel
//				technologies.
//
// Copyright (c) 2020 by Tecnologico de Monterrey.
// All Rights Reserved. May be reproduced for any non-commercial
// purpose.
//
// =================================================================

public class Example7 {
	private static final int SIZE = 1_000_001;
	private boolean array[];

	public Example7() {
		this.array = array;
	}

	// place yout code here

	public void calculate() {
		// place yout code here
	}

	public static void main(String args[]) {
		long startTime, stopTime;
		double result, ms;
		Example7 obj;

		ms = 0;
		result = 0;
		obj = new Example7();
		System.out.printf("Starting...\n");
		for (int i = 0; i < Utils.N; i++) {
			startTime = System.currentTimeMillis();

			// Call yout method here.

			stopTime = System.currentTimeMillis();

			ms += (stopTime - startTime);
		}
		System.out.printf("sum = %.0f\n", result);
		System.out.printf("avg time = %.5f ms\n", (ms / Utils.N));
	}
}

File: examples/02-intro-cpp/example7.cpp
// =================================================================
//
// File: example7.cpp
// Author: Pedro Perez
// Description: This file contains the code to brute-force all
//				prime numbers less than MAXIMUM. The time this
//				implementation takes will be used as the basis to
//				calculate the improvement obtained with parallel
//				technologies.
//
// Copyright (c) 2020 by Tecnologico de Monterrey.
// All Rights Reserved. May be reproduced for any non-commercial
// purpose.
//
// =================================================================

#include <iostream>
#include <iomanip>
#include <cstring>
#include <cmath>
#include <algorithm>
#include "utils.h"

#define MAXIMUM 5000000 //5e6

using namespace std;

// implement your class here

int main(int argc, char* argv[]) {
	int i;
	double ms;

	cout << "Starting..." << endl;
	ms = 0;
	// create object here
	for (int i = 0; i < N; i++) {
		start_timer();

		// call your method here.

		ms += stop_timer();
	}
	cout << "result = " << setprecision(2) << "???";
	cout << "avg time = " << setprecision(15) << (ms / N) << " ms" << endl;

	delete [] a;
	return 0;
}
